week3/homework/src/utilities.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const dataDealer = require('./data/data_dealer');
+const read = dataDealer.read;
+const write = dataDealer.write;
+const shortid = require('shortid');
+const TodoItem = require('./todo_obj_constructor');
+
+const validation = function(req) {
+  const reqBodyLength = Object.keys(req.body).length;
+  const reqToDo = req.body.todo || {};
```

Comment:
instead of all this we can do the following:

```javascript
const reqBodyLength = Object.keys(req.body).length
let reqDescription = reqToDo.description

if (reqBodyLength) {
  reqDescription = reqDescription.trim()
} else {
  reqDescription = ''
}
return reqDescription // your function and this if statement work the same but shorter version.
```

Some comments on this function may approve my thought:
 - line 19: what if the user sent accidentally another key?
 - line 20: these can't be equal! and always fail. please look at this article about [object equality]
 - lines 21 and 22 are a correct condition in your case.

[object equality]: http://adripofjavascript.com/blog/drips/object-equality-in-javascript.html

Reply:
- about your point line 19: if the user sends 2 ToDo object :
`{"todo:{"description":"task 1"} ,{"todo:{"description":"task 2"}`}`
here something wired going to happen: first that is not valid JSON, duplicated keys. Second the both ToDos wil be added correctly to the data.

- About your point line 20: in line 11 and 12 I assign the value `{}` to `reqDescription` when something is not defined. That is something wrong I made. I forget the concept of the object in the memory. Using `null` instead of `{}` can solved this issue. 

- Thank for writing a shorter code , what makes my code longer is : I tried to deal with a situation like this: 
`{"todoes:{"description":"task 1"}` so if the key is something else that `todo`, and my way makes things more complicated. 

- object equality: is something I really  need to review . Thanks